pkg/providers/gcpsecrets.go

```diff
@@ -0,0 +1,59 @@
+package providers
+
+import (
+	"context"
+	"fmt"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	"google.golang.org/api/option"
+)
+
+type GoogleSecrets struct {
+	project string
+	client  *secretmanager.Client
+}
+
+func NewGoogleSecretsProviderFromEnv() (GoogleSecrets, error) {
+	credentials, err := LookupEnv(GoogleSecretsAccountCredentials)
+	if err != nil {
+		return GoogleSecrets{}, err
+	}
+
+	project, err := LookupEnv(GoogleSecretsProject)
+	if err != nil {
+		return GoogleSecrets{}, err
+	}
+	return NewGoogleSecretsProvider(credentials, project)
+}
+
+func NewGoogleSecretsProvider(credentials, project string) (GoogleSecrets, error) {
+	ctx := context.Background()
+
+	client, err := secretmanager.NewClient(ctx,
+		option.WithCredentialsFile(credentials),
+	)
+	if err != nil {
+		return GoogleSecrets{}, err
+	}
+
+	return GoogleSecrets{
+		project: project,
+		client:  client,
+	}, nil
+}
+
+func (gs GoogleSecrets) Close() error {
+	return gs.client.Close()
+}
+
+func (gs GoogleSecrets) GetSecret(path string) (value string, err error) {
```

Comment:
name, since you for the path later, from it

Reply:
Actually for consistency reasons (everywhere else) i'd rather keep it as path since that's the convention in the interface.

Nice catch tho!